backend/tilavarauspalvelu/management/commands/create_robot_test_data.py

```diff
@@ -0,0 +1,2443 @@
+    ReservationKind,
+    ReservationNotification,
+    ReservationStartInterval,
+    ReservationStateChoice,
+    ReservationTypeChoice,
+    Weekday,
+)
+from tilavarauspalvelu.models import (
+    Application,
+    ApplicationRound,
+    Equipment,
+    OriginHaukiResource,
+    PaymentAccounting,
+    Purpose,
+    Reservation,
+    ReservationMetadataSet,
+    ReservationPurpose,
+    ReservationSeries,
+    ReservationUnit,
+    ReservationUnitCancellationRule,
+    ReservationUnitType,
+    Space,
+    TaxPercentage,
+    TermsOfUse,
+    Unit,
+    User,
+)
+from tilavarauspalvelu.typing import TimeSlotDB
+from utils.date_utils import local_date, local_datetime
+
+from tests.factories import (
+    ApplicationRoundFactory,
+    ApplicationRoundTimeSlotFactory,
+    ReservationFactory,
+    ReservationUnitFactory,
+    UnitFactory,
+)
+
+if TYPE_CHECKING:
+    from collections.abc import Generator
+
+__all__ = [
+    "create_robot_test_data",
+]
+
+
+class Command(BaseCommand):
+    help = "Creates test data for robotframework tests."
+
+    def handle(self, *args: Any, **options: Any) -> None:
+        create_robot_test_data()
+
+
+@contextmanager
+def handle_model_object_errors() -> Generator[None]:
+    try:
+        yield
+
+    except ObjectDoesNotExist as error:
+        msg = f"Object does not exist: {error}"
+        raise ValidationError(msg, code="object_does_not_exist") from error
+
+    except MultipleObjectsReturned as error:
+        msg = f"Multiple objects returned: {error}"
+        raise ValidationError(msg, code="multiple_objects_returned") from error
+
+
+@transaction.atomic()
+@handle_model_object_errors()
+def create_robot_test_data() -> None:
+    if os.getenv("DJANGO_SETTINGS_ENVIRONMENT") == "Production":
+        msg = "Hey! This is the production environment! Don't just try to generate test data! >:("
+        raise ValidationError(msg, code="production_environment")
+
+    remove_existing_data()
+    create_users()
+    create_reservation_units()
+    create_application_rounds()
+    create_past_reservations()
+
+
+def remove_existing_data() -> None:
+    try:
+        harakka = Unit.objects.get(name="Harakka, piilokoju")
+
+    # If Harakka doesn't exist, then none of the other entities should exist either.
+    except Unit.DoesNotExist:
+        return
+
+    application_round = ApplicationRound.objects.filter(name="Kausivaraus (AUTOMAATIO TESTI ÄLÄ POISTA)")
+
+    reservation_units = ReservationUnit.objects.filter(unit=harakka)
+    spaces = Space.objects.filter(unit=harakka)
+
+    sq = Subquery(reservation_units.values("id"))
+    reservations = Reservation.objects.filter(reservation_unit__in=sq)
+    series = ReservationSeries.objects.filter(reservation_unit__in=sq)
+    applications = Application.objects.filter(
+        models.Q(application_round__reservation_units__in=sq)
+        | models.Q(application_round__name="Kausivaraus (AUTOMAATIO TESTI ÄLÄ POISTA)")
+    )
+
+    spaces.delete()
```

Comment:
and `resources` ?

Reply:
No resourcs in any of the reservation units